inspirehep/modules/theme/templates/inspirehep_theme/format/record/Inspire_HTML_detailed_macros.tpl

```diff
@@ -119,14 +119,11 @@
         {% endif %}
     {% endfor %}
 
-    {% if record.free_keywords or record.thesaurus_terms|length > 10 %}
+    {% if record.free_keywords or record.keywords|length > 10 %}
       <div>
         <a href="" class="text-muted" data-toggle="modal" data-target="#keywordsFull">
           <small>
-            Show all {{ record.thesaurus_terms|length }} keywords
-            {% if record.get('free_keywords') %}
-              plus author supplied keywords
-            {% endif %}
```

Comment:
Now author supplied and thesaurus keywords are in the same property, I've merged them in the view because it's easier, but I can add logic to show them separated if need be, just let me know.